File: src/main/java/dev/esophose/discordbot/command/commands/misc/EmbedCommand.java
package dev.esophose.discordbot.command.commands.misc;

import dev.esophose.discordbot.Sparky;
import dev.esophose.discordbot.command.DiscordCommand;
import dev.esophose.discordbot.command.DiscordCommandMessage;
import dev.esophose.discordbot.manager.CommandManager;
import discord4j.core.object.entity.channel.GuildMessageChannel;
import discord4j.core.object.util.Permission;
import discord4j.core.object.util.PermissionSet;
import java.util.Collections;
import java.util.List;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class EmbedCommand extends DiscordCommand {

    public EmbedCommand(Sparky bot) {
        super(bot);
    }

    public void execute(DiscordCommandMessage message, String title, String text) {
        message.delete().subscribe();

        if (title.equals("null"))
            title = null;

        text = text.replaceAll(Pattern.quote("\\n"), Matcher.quoteReplacement("\n"));

        this.bot.getManager(CommandManager.class).sendResponse(message.getChannel(), title, text).subscribe();
    }

    @Override
    public String getName() {
        return "embed";
    }

    @Override
    public List<String> getAliases() {
        return Collections.emptyList();
    }

    @Override
    public String getDescription() {
        return "Makes the bot say a message";
    }

    @Override
    public PermissionSet getRequiredBotPermissions() {
        return PermissionSet.of(Permission.SEND_MESSAGES);
    }

    @Override
    public Permission getDefaultRequiredMemberPermission() {
        return Permission.ADMINISTRATOR;
    }

}

File: src/main/kotlin/dev/esophose/discordbot/command/commands/misc/EmbedCommand.kt
package dev.esophose.discordbot.command.commands.misc

import dev.esophose.discordbot.Sparky
import dev.esophose.discordbot.command.DiscordCommand
import dev.esophose.discordbot.command.DiscordCommandMessage
import dev.esophose.discordbot.manager.CommandManager
import discord4j.rest.util.Permission
import discord4j.rest.util.PermissionSet
import java.util.regex.Matcher
import java.util.regex.Pattern

class EmbedCommand : DiscordCommand() {

    override val name: String
        get() = "embed"

    override val aliases: List<String>
        get() = emptyList()

    override val description: String
        get() = "Makes the embed a message"

    override val requiredBotPermissions: PermissionSet
        get() = PermissionSet.of(Permission.SEND_MESSAGES)

    override val defaultRequiredMemberPermission: Permission
        get() = Permission.ADMINISTRATOR

    fun execute(message: DiscordCommandMessage, title: String, text: String) {
        message.delete().subscribe()

        val embedTitle = if (title == "null") null else title
        val embedText = text.replace(Pattern.quote("\\n").toRegex(), Matcher.quoteReplacement("\n"))
        Sparky.getManager(CommandManager::class).sendResponse(message.channel, embedTitle, embedText).subscribe()
    }

}
